lib/compat/wordpress-6.5/class-wp-navigation-block-renderer.php

```diff
@@ -544,22 +544,26 @@ private static function get_custom_overlay_blocks_html( $attributes ) {
 			);
 			if ( $overlay_template_parts ) {
 				$content = $overlay_template_parts[0]->content;
-				$blocks = [];
-				foreach( parse_blocks( $content ) as $block ) {
-					if ( 'core/navigation' === $block['blockName'] ) {
-						$block['attrs']['overlayMenu'] = "never";
-						if ( ! empty( $block['attrs']['ref'] ) && empty( $attributes['ref'] ) ) {
-							$block['attrs']['ref'] = $attributes['ref'];
-						}
-					}
-					$blocks[] = $block;
-				}
-				$content = serialize_blocks( $blocks );
 			}
 		} else {
 			$content = $custom_overlay_post->post_content;
 		}
 
+
+		// Update the ref on the block
+		// if it doesn't have one, and turn off the overlay.
```

Comment:
Why?

Reply:
1. Update the ref:
To enable themes to provide a navigation-overlay.html. The theme won't know what ref to give the block so we have to pass it from the parent block

2. Turn off the overlay
Navigation blocks inside an overlay can't also have an overlay otherwise we get into an infinite loop of overlays